execution chapter (#25)
mqtt-varroa/varroa-documentation: LaTeX/Parts/Chapter-Execution.tex

\chapter{Varroa Execution}

> Varroa has two main modes of execution as of now. Either execute it directly and configure as necessary manually / programmatically ahead of time, or use containers and utilize dynamic discovery.

Reviewer: ```suggestion
Varroa has two main modes of execution for now. One is to configure Varroa ahead of time manually or programmatically and then execute directly. The other is to use containers and utilize dynamic discovery.
```

\section{Configuration}

> Following are the two sequence sketches for direct (static initialization) configuration and dynamic (DNS discovery based) configuration.

Reviewer: reference links missing

\section{Static configuration}
\begin{figure}[h]
\begin{center}
\includegraphics[scale=0.65]{Resources/PDF/ExecutionStaticInit}
\caption{static execution}
\end{center}
\end{figure}

Static configuration may be used on any infrastructure where hostname/ip addresses of all Varroa instances (commander and agents) are known ahead of time. This is the case on most typical systems, such as IaaS clouds where an instance is booted and assigned an address before the application is deployed onto that system.

> For static configuration, the user must provide an instance configuration, which configures the Varroa instance as either a commander or an agent, as well as (optionally) the instance limits specifying how much load the agent can generate.

Reviewer: This is wrong, the configuration can specify both agent and commander,

\section{Dynamic configuration}

\begin{figure}[h]
\begin{center}
\includegraphics[scale=0.65]{Resources/PDF/ExecutionDnsInit}
\caption{dynamic execution}
\end{center}
\end{figure}

> DNS discovery specifically is aimed at container environments and cloud platforms. These types of platforms generally provide different methods of service discovery, which are necessary to allow distributed systems to communicate with their components.

Reviewer: ```suggestion
DNS discovery specifically is aimed at container environments and cloud platforms. These types of platforms generally provide different methods of service discovery, which are necessary to allow the parts of a distributed systems to communicate with each other.
```
